src/pages/index.js

```diff
@@ -0,0 +1,331 @@
+  const cardDeleteBtn = cardElement.querySelector(".card__trash-button");
+
+  cardLikedBtn.addEventListener("click", () => {
+    if (cardLikedBtn.classList.contains("card__like-button_liked")) {
+      api
+        .nonlikedCard(data._id)
+        .then(() => {
+          cardLikedBtn.classList.remove("card__like-button_liked");
+        })
+        .catch(console.error);
+    } else {
+      api
+        .likeCard(data._id)
+        .then(() => {
+          cardLikedBtn.classList.add("card__like-button_liked");
+        })
+        .catch(console.error);
+    }
+  });
+
+  cardDeleteBtn.addEventListener("click", () =>
+    handleCardDelete(cardElement, data)
+  );
+
+  image.addEventListener("click", () => {
+    openModal(previewModal);
+    preModalCaptionEl.textContent = data.name;
+    preModalImage.src = data.link;
+    preModalImage.alt = data.name;
+  });
+
+  cardNameEl.textContent = data.name;
+  image.src = data.link;
+  image.alt = data.alt;
+
+  if (data.isLiked) {
+    cardLikedBtn.classList.add("card__like-button_liked");
+  }
+  return cardElement;
+}
+
+function handleDeleteSubmit() {
+  // evt.preventDefault()
+  setSubmitButtonText(deleteModalDelBtn, "Deleting");
+
+  api
+    .deleteCard(selectedCardId)
+    .then(() => {
+      selectedCard.remove();
+      closeModal(cardDeleteModal);
+      newCardForm.reset();
+    })
+    .catch(console.error)
+    .finally(() => {
+      setSubmitButtonText(deleteModalDelBtn, "Delete");
+    });
+}
+function handleCardDelete(cardElement, data) {
+  selectedCard = cardElement;
+  selectedCardId = data._id;
+  openModal(cardDeleteModal);
+}
+deleteModalDelBtn.addEventListener("click", handleDeleteSubmit);
+deleteModalCanclBtn.addEventListener("click", () => {
+  closeModal(cardDeleteModal);
+});
+
+function openModal(modal) {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscKeyPress);
+  document.addEventListener("click", handleOverlay);
+}
+
+function closeModal(modal) {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscKeyPress);
+  document.removeEventListener("click", handleOverlay);
+}
+
+function handleEscKeyPress(evt) {
+  if (evt.key === "Escape") {
+    const openedModal = document.querySelector(".modal_opened");
+    if (openedModal) {
+      closeModal(openedModal);
+    }
+  }
+}
+
+function handleEditFormSubmit(evt) {
+  evt.preventDefault();
+  const cardModalSubmitBtn = evt.submitter;
+  cardModalSubmitBtn.textContent = "Saving...";
+
+
+
+  api
+    .editUserInfo({
+      name: editModalProfileName.value,
+      about: editModalProfileDescription.value,
+    })
+    .then((data) => {
+      debugger;
+      profileDescriptionElement.textContent = data.about;
+      profileNameElement.textContent = data.name;
+      closeModal(editProfileModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      cardModalSubmitBtn.textContent = "Save";
```

Comment:
**COULD BE IMPROVED**

If it’s interesting for you here is how we can make a universal function for handling any submit. We can get rid of such duplicating as **loading effect**, **resetting** and **catching errors**


```js
// define a function for changing the button text. It accepts 4 params (the 2 last are optional with default texts)
export function renderLoading(isLoading, button, buttonText='Save', loadingText='Saving...') {
  if (isLoading) {
    button.textContent = loadingText
  } else {
    button.textContent = buttonText
  }
}

// define a universal function that accepts a request function, event and a default loading text 
function handleSubmit(request, evt, loadingText = 'Saving...') {
 // You need to prevent the default action in any submit handler
  evt.preventDefault();

  // the button is always available inside `event` as `submitter`
  const submitButton = evt.submitter;
  // fix the initial button text
  const initialText = submitButton.textContent;
  // change the button text before requesting
  renderLoading(true, submitButton, initialText, loadingText);
  // call the request function to be able to use the promise chain
  request()
    .then(() => {
      // any form should be reset after a successful response 
      // evt.target is the form in any submit handler		
      evt.target.reset();
    })
      // we need to catch possible errors
      // console.error is used to handle errors if you don’t have any other ways for that
    .catch(console.error)      

    // and in finally we need to stop loading
    .finally(() => {
      renderLoading(false, submitButton, initialText);
    });
}

```

Here is an example of handling the profile form submit:

```js
function handleProfileFormSubmit(evt) {
  // create a request function that returns a promise
  function makeRequest() {
    // `return` lets us use a promise chain `then, catch, finally`
    return editProfile(nameInput.value, jobInput.value).then((userData) => {
      userName.textContent = userData.name;
      userJob.textContent = userData.about;
    });
  }
  // here we call handleSubmit passing the request and event (if you want a different loading text then you need to pass the 3rd param)
  handleSubmit(makeRequest, evt);
}

```

So, this way you can remove a lot of code duplicating. You will not need to search buttons, pass initial button texts and so on.

`handleSubmit` and `renderLoading` should be placed in `utils.js`, because they are utility functions